app/controlplane/internal/service/service.go

```diff
@@ -196,6 +206,64 @@ func (s *service) userHasPermissionOnProject(ctx context.Context, orgID string,
 	return s.authorizeResource(ctx, policy, authz.ResourceTypeProject, p.ID)
 }
 
+// userHasPermissionOnGroupMemberships is a helper function that checks if a user has permission on a group or group membership.
+// If the user is a maintainer of the group, they have permission regardless of their role
+// If the user has an admin or owner role in the organization, they have permission
+// Otherwise, the operation is forbidden
+func (s *service) userHasPermissionOnGroupMemberships(ctx context.Context, orgID string, groupIdentifier *pb.IdentityReference) error {
```

Comment:
let's review this tomorrow with @jiparis to see how this fits in the overall RBAC story. 

It seems yet another RBAC exception outside of casbin that could be potentially implemented using the new RBAC methods